### ISSUE TITLE
"helm/resource-policy" was set to keep when installing certain helm charts via fleet, for unknown reason.

### ISSUE THREAD

**Issue by aDisplayName** [rancher/fleet#2716]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When using fleet to deploy the following fleet charts, workload resources are annotated with `helm/resource-policy: keep`, but we could not figure out why. By searching the source code of their helm charts, we can only see the crds are annotated with such setting.
| app | helm repo | helm charts | GitHub repository |
|-|-|-|-|
|argo workflows| https://argoproj.github.io/argo-helm | argo-workflows:0.41.14 | https://github.com/argoproj/argo-helm/tree/main/charts/argo-workflows |
| cert-manager | https://charts.jetstack.io | cert-manager:v1.15.2 | https://github.com/cert-manager/cert-manager/tree/master/deploy/charts/cert-manager | 
| bitnami/cert-manager |  oci://registry-1.docker.io/bitnamicharts/cert-manager | :1.3.16| https://github.com/bitnami/charts/blob/main/bitnami/cert-manager/ |

Using `argo-workflows:0.41.14` as an example, the only resources annotated with `helm.sh/resource-policy: keep` are those under `/templates/crds`: https://github.com/search?q=repo%3Aargoproj%2Fargo-helm+path%3A%2Fcharts%2Fargo-workflows%2F**+%22resource-policy%22&type=code

But when installing via fleet config: 
```yaml
defaultNamespace: argo

helm:
  # Use a custom location for the Helm chart. This can refer to any go-getter URL.
  # This allows one to download charts from most any location.  Also know that
  # go-getter URL supports adding a digest to validate the download. If repo
  # is set below this field is the name of the chart to lookup
  chart: argo-workflows

  repo: https://argoproj.github.io/argo-helm

  version: 0.41.14
  releaseName: argo-workflows
  values:
    fullnameOverride: argo-workflows
    server:
      extraArgs: [--auth-mode=server]
    images:
      pullPolicy: IfNotPresent
    crds:
      keep: false # avoid argo conflicts between different gitrepo due to ownership annotation
```

The resulted deployments are annotated with `helm.sh/resource-policy` set to `keep`
```yaml
metadata:
  name: argo-workflows-server
  annotations:
    deployment.kubernetes.io/revision: '1'
    helm.sh/resource-policy: keep
    meta.helm.sh/release-name: argo-workflows
    meta.helm.sh/release-namespace: argo
    objectset.rio.cattle.io/id: default-resource-policy-bug-app-argo
#    key: string
  creationTimestamp: '2024-08-08T15:03:50Z'
  generation: 1
  labels:
    app: server
    app.kubernetes.io/component: server
    app.kubernetes.io/instance: argo-workflows
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: argo-workflows-server
    app.kubernetes.io/part-of: argo-workflows
    app.kubernetes.io/version: v3.5.10
    helm.sh/chart: argo-workflows-0.41.14
    objectset.rio.cattle.io/hash: 9755058833ea56cf90586e04fe3b080a49eca171
#    key: string
  namespace: argo
  resourceVersion: '8481'
  uid: ca40a5bb-5d94-4c5c-88b1-06e68bc6b8b9
  fields:
    - argo-workflows-server
    - 1/1
    - 1
    - 1
    - 20s
    - argo-server
    - quay.io/argoproj/argocli:v3.5.10
    - >-
      app.kubernetes.io/instance=argo-workflows,app.kubernetes.io/name=argo-workflows-server
```

### Expected Behavior

The workloads are not annotated with `helm.sh/resource-policy: keep` unless it is defined in helm charts.

### Steps To Reproduce

The sample fleet yaml config can be located at https://github.com/aDisplayName/bugsamplecode/tree/main/20240808

### Environment

```markdown
- Architecture:
- Fleet Version: 0.9.6
- Rancher version: 2.8.4
- Cluster:
  - Provider: k3s
  - Kubernetes Version: 1.21.14, 1.28.12
```

**Comment by aDisplayName** [comment]
keep it alive

**Comment by guozijing** [comment]
also meet this issue, any updates?

**Comment by guozijing** [comment]
> keep it alive

I found if the helm charts contain crd, then all the resources will be added the helm/resource-policy, when I delete the crds from the helm charts, it will be okay.

**Comment by aDisplayName** [comment]
@manno , Is there a plan to backport the fix to fleet v0.9?

**Comment by kkaempf** [comment]
> @manno , Is there a plan to backport the fix to fleet v0.9?

No, not currently.

**Comment by sbulage** [comment]
### System Information

| Rancher Version | Fleet Version |
|--------|--------|
| 2.10.0-alpha5 | fleet:v0.11.0-beta.3 | 

Followed below scenario's considering the `deleteCRDResources` values (`true/false`) and `helm.sh/resource-policy: keep` annotation behavior.

---

### Scenario 1
- Create a `GitRepo` which installs CRD's along with the application.(Used this [fleet.yaml](https://github.com/sbulage/test-fleet/blob/main/argo-crds/fleet.yaml) file.)
- Make sure `deleteCRDResources: true` must be present in the `fleet.yaml`.
- Verified that annotation `helm.sh/resource-policy: keep` is _absent_.

---

### Scenario 2
- Create a `GitRepo` which installs CRD's along with the application.(Used this [fleet.yaml](https://github.com/sbulage/test-fleet/blob/main/argo-crds/fleet.yaml) file.)
- Update the value of `deleteCRDResources: false` in the `fleet.yaml`.
- Verified that annotation `helm.sh/resource-policy: keep` is _present_.

### Point to consider:
_If we don't want to see annotation `helm.sh/resource-policy: keep` then we have to use the `deleteCRDResource` option with `true` value in `fleet.yaml`._

---

### Annotation on CRD

| Scenario 1 | Scenario 2 |
|--------|--------|
| `deleteCRDResource`: `true` | `deleteCRDResource`: `false` |
| ![Image](https://github.com/user-attachments/assets/5b19d39c-32c9-4cb6-ac93-d828cfb634c8) | ![Image](https://github.com/user-attachments/assets/36191127-02b6-415e-b0eb-8e9b29323416) | 

---

### Screenshots:
<details>
<summary><b>Annotation is absent on NonCRD resources(ConfigMaps, Deployment, Service Account, Service).</b></summary>

![Image](https://github.com/user-attachments/assets/c5344162-3eb3-456a-8a02-da71517e4900)


</details>
